sympy/core/numbers.py

```diff
@@ -1691,6 +1691,47 @@ def limit_denominator(self, max_denominator=1000000):
         f = fractions.Fraction(self.p, self.q)
         return Rational(f.limit_denominator(fractions.Fraction(int(max_denominator))))
 
+    def limit_div(self, a, b = None):
```

Comment:
I think this can just be like the following
```python
def limit_numerator(self, n):
    if not n or not self:
        return S.Zero
    if self < 0:
        return -(-self).limit_numerator(n)
    return 1/(1/self).limit_denominator(n)
```
Note that for the doctest example this gives a better answer:
```
>>> r
1819831678027117/9007199254740992
>>> a
11003/54459
>>> b = 1/(1/r).limit_denominator(10**5); b
76269/377491
>>> abs(b-r) < abs(a-r)
True
```
So I'm not sure where I went wrong, but the current routine is not right.

Reply:
Also, I wonder if an improper fraction of a number greater than 1 should have the numerator of the fractional part reduced. Consider:
```python
>>> p=Rational(pi.n())
>>> p
884279719003555/281474976710656
>>> r=p-int(p)
>>> r
39854788871587/281474976710656

For this case they are the same

>>> 1/(1/p).limit_denominator(1000)
355/113
>>> 3 + 1/(1/r).limit_denominator(1000)
355/113


But not for this case

>>> a=1/(1/p).limit_denominator(10**7)
>>> b=3+1/(1/r).limit_denominator(10**7)
>>> abs(a-p)=<abs(b-p)
False
>>> a,b
(5419351/1725033, 165707065/52746197)
```